### PR TITLE
Update Rubocop version & fix offenses

### PR DESCRIPTION
**Pull request by archanaserver**
This PR updates the RuboCop version and addresses various code style offenses.

### CODE REVIEW

File: bin/foreman-proxy-certs-generate [hunk 1]
```diff
@@ -16,10 +16,10 @@ Kafo::KafoConfigure.hooking.register_pre(:init) do
...
       scenario = YAML.load_file(data[:answer_file])
       organization = begin
-                       scenario['foreman']['initial_organization']
-                     rescue StandardError
-                       nil
-                     end
+        scenario['foreman']['initial_organization']
+      rescue StandardError
+        nil
+      end
```
Comment (ekohl) [inline bin/foreman-proxy-certs-generate]: You can use [Hash#dig](https://www.rubydoc.info/stdlib/core/Hash:dig) together with the same navigator operator to safely get the value:
```suggestion
      organization = scenario&.dig('foreman', 'initial_organization')
```

File: hooks/boot/10-reset_data.rb [hunk 2]
```diff
@@ -1,10 +1,7 @@
 app_option(
   '--reset-data',
   :flag,
-  'This option will drop all databases for Foreman and subsequent backend systems. ' +
-  "You will lose all data!\nUnfortunately, we " +
-  "can't detect a failure, so you should verify success " +
-  "manually.\nDropping can fail when the DB is in use.",
+  "This option will drop all databases for Foreman and subsequent backend systems. You will lose all data!\nUnfortunately, we can't detect a failure, so you should verify success manually.\nDropping can fail when the DB is in use.",
```
Comment (ekohl) [inline hooks/boot/10-reset_data.rb]: I'm not sure such a long line is an improvement.

File: hooks/boot/01-kafo-hook-extensions.rb [hunk 3]
```diff
@@ -85,7 +85,7 @@ def pulpcore_enabled?
     module_enabled?('foreman_proxy_content')
   end
 
-  def log_and_say(level, message, do_say = true, do_log = true)
+  def log_and_say(level, message, do_say: true, do_log: true)
```
Comment (ekohl) [inline hooks/boot/01-kafo-hook-extensions.rb]: This is an API change and I don't see all invocations of `log_and_say` changed

File: .rubocop.yml [hunk 4]
```diff
@@ -126,3 +126,33 @@ Style/NumericPredicate:
 
 Layout/HeredocIndentation:
   Enabled: false
+
+Style/SoleNestedConditional:
```
Comment (ekohl) [inline .rubocop.yml]: I'd prefer to use `NewCops` under `AllCops` to enable cops.

File: spec/puppet_version_spec.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 require 'json'
 require 'semantic_puppet'
 
-SUPPORTED_PUPPET_VERSIONS = ['7.0.0'].map { |v| SemanticPuppet::Version.parse(v) }
+VERSIONS = ['7.0.0'].map { |v| SemanticPuppet::Version.parse(v) }
```
Comment (ekohl) [inline spec/puppet_version_spec.rb]: Why rename this? The previous name was descriptive.

File: spec/puppet_version_spec.rb [hunk 6]
```diff
@@ -15,8 +15,8 @@
         SemanticPuppet::VersionRange.parse(version_requirement)
       end
 
-      SUPPORTED_PUPPET_VERSIONS.each do |puppet_version|
-        it "supports Puppet #{puppet_version}" do
+      VERSIONS.each do |puppet_version|
+        it "should support Puppet #{puppet_version}" do
```
Comment (ekohl) [inline spec/puppet_version_spec.rb]: Isn't this quite the other way around of what RuboCop suggests? It literally reverts 31782b64031ad73afa5a0349d52891a633587594, which was flagged by rubocop-rspec.

File: Gemfile [hunk 7]
```diff
@@ -14,7 +14,7 @@ gem 'rake'
 
 group :test do
   gem 'rspec'
-  gem 'rubocop', '~> 0.80.0'
+  gem 'rubocop', '~> 1.56.0'
```
Comment (ekohl) [inline Gemfile]: Can you also add `rubocop-rake`, `rubocop-rspec` and `rubocop-performance`?

**Review by ekohl — CHANGES_REQUESTED** [review]

File: hooks/boot/05-environment.rb [hunk 8]
```diff
@@ -1,5 +1,5 @@
 %w[http_proxy https_proxy ssl_cert_file HTTP_PROXY HTTPS_PROXY SSL_CERT_FILE].each do |variable|
-  if ::ENV.delete(variable)
+  if ENV.delete(variable)
```
Comment (ekohl) [inline hooks/boot/05-environment.rb]: RuboCop is wrong here, and it needs to be `::ENV` so it's not interpreted as `Kafo::ENV`. See https://github.com/theforeman/kafo/blob/a1b1b12d47f2e0dd26629452809c915e40a861b4/lib/kafo/kafo_configure.rb#L3-L8

Comment (archanaserver) [reply hooks/boot/05-environment.rb]: @ekohl Thanks! I've fixed it now and it is ready for the review.

File: hooks/pre_commit/05-puppet_certs_exist.rb [hunk 9]
```diff
@@ -8,8 +8,8 @@
 
 if puppet_ca_enabled && key_path && File.exist?(key_path) && !kafo.skip_checks_i_know_better?
   if !puppet_ca_cert.empty? && !File.exist?(puppet_ca_cert)
-    fail_and_exit("The file #{puppet_ca_cert} does not exist.\n #{client_message}\n" \
-    " - if you use custom Puppet SSL directory (--foreman-proxy-ssldir) make sure the directory exists and contain the CA certificate.\n", 101)
+    fail_and_exit("The file #{puppet_ca_cert} does not exist.\n #{client_message}\n " \
+                  "- if you use custom Puppet SSL directory (--foreman-proxy-ssldir) make sure the directory exists and contain the CA certificate.\n", 101)
```
Comment (ekohl) [inline hooks/pre_commit/05-puppet_certs_exist.rb:12]:
```suggestion
    fail_and_exit("The file #{puppet_ca_cert} does not exist.\n #{client_message}\n" \
                  " - if you use custom Puppet SSL directory (--foreman-proxy-ssldir) make sure the directory exists and contain the CA certificate.\n", 101)
```

File: .rubocop.yml [hunk 10]
```diff
@@ -1,12 +1,101 @@
...
+  Enabled: true
+
+Style/RedundantConstantBase:
+  Enabled: true
+
+Style/FileWrite:
+  Enabled: true
+
+Layout/LineEndStringConcatenationIndentation:
+  Enabled: true
```
Comment (ekohl) [inline .rubocop.yml]: I think these redundant

File: hooks/pre_commit/05-puppet_certs_exist.rb (hunk 9 above)
Comment (archanaserver) [reply hooks/pre_commit/05-puppet_certs_exist.rb:12]: Making these changes, raise two new offenses Layout/LineEndStringConcatenationIndentation and Layout/LineContinuationLeadingSpace. Should I disable it inline? @ekohl

File: hooks/boot/01-kafo-hook-extensions.rb (hunk 3 above)
Comment (archanaserver) [reply hooks/boot/01-kafo-hook-extensions.rb]: @ekohl it is changed now!

Comment (archanaserver) [reply hooks/boot/01-kafo-hook-extensions.rb]: @ekohl I think this changes still needs to be review?

File: spec/migration_spec.rb [hunk 11]
```diff
@@ -135,7 +135,7 @@
         }
       end
 
-      it 'changes the default to Redis' do
+      it 'with changes the default to Redis' do
```
Comment (ekohl) [inline spec/migration_spec.rb]: This doesn't read well
```suggestion
      it 'changes the default to Redis' do
```

File: .rubocop.yml [hunk 12]
```diff
@@ -1,12 +1,59 @@
...
 
+Style/StringConcatenation:
+  Enabled: true
+  Exclude:
+    - 'hooks/boot/10-reset_data.rb'
+
+Style/SlicingWithRange:
+  Enabled: false
+
+Performance/Casecmp:
```
Comment (ekohl) [inline .rubocop.yml]: Why disable these performance cops? Many of these look more like something you'd do in a TODO file.

The file `.rubocop.yml` is what you use as a set of rules that you've agreed on. The file `.rubocop_todo.yml` is the file for rules that currently don't fit the guidelines, but that you want to eventually.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: .rubocop.yml (hunk 12 above)
Comment (archanaserver) [reply .rubocop.yml]: Previously, I disabled the `Performance/Casecmp` cop due to safety concerns as mentioned in the documentation (https://docs.rubocop.org/rubocop-performance/cops_performance.html#performancecasecmp). Upon reviewing the code changes when enabling this cop, it seems to be safe in this case.

I also observed that enabling the cop resulted in a `Style/NegatedIf` offense, which I disabled to ensure consistency in our code style.

**Comment by dosas** [comment]
I would also require an up to date rubocop. What do we need to do to get this forward?

File: .rubocop.yml [hunk 13]
```diff
@@ -1,12 +1,56 @@
...
+  Enabled: false
+
+RSpec/NestedGroups:
+  Enabled: false
+
+RSpec/ExampleLength:
+  Enabled: false
+
+RSpec/ExpectInHook:
+  Enabled: false
```
Comment (ekohl) [inline .rubocop.yml]: Please keep these as TODOs and don't disable them outright.

File: .rubocop.yml [hunk 14]
```diff
@@ -126,3 +170,12 @@ Style/NumericPredicate:
...
   Enabled: false
+
+Style/OptionalBooleanParameter:
+  Enabled: false
+
+Style/NegatedIf:
+  Enabled: false
+
+RSpec/ContextWording:
+  Enabled: false
```
Comment (ekohl) [inline .rubocop.yml]: Same for these

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by dosas** [comment]
@archanaserver Could you implement the changes?

**Comment by archanaserver** [comment]
> @archanaserver Could you implement the changes?

@dosas I recommend handling these changes through a separate pull request. Alternatively, we could consider incorporating them when the theforeman-rubocop gem releases its next version with the updated RuboCop version. You can find more details about this here:  https://community.theforeman.org/t/standardizing-rubocop-with-theforeman-rubocop/37239 Let me know your thoughts!

**Comment by dosas** [comment]
> > @archanaserver Could you implement the changes?
> 
> @dosas I recommend handling these changes through a separate pull request. Alternatively, we could consider incorporating them when the theforeman-rubocop gem releases its next version with the updated RuboCop version. 

I don't understand AFAIK you already implemented the requested changes in a commit?

>You can find more details about this here: https://community.theforeman.org/t/standardizing-rubocop-with-theforeman-rubocop/37239 Let me know your thoughts!

LGTM

**Review by ehelms — APPROVED** [review]
I think what's present in the PR is good and makes solid updates to update Rubocop. @ekohl I'd like to merge this in the next day or two unless you have further comments that can't be addressed in a follow up.

**Comment by ehelms** [comment]
[test foreman-installer]

**Pull request merged**